kumascript/macros/js_property_attributes.ejs

```diff
@@ -9,104 +9,103 @@
+    writableName: "Writable",
+    enumerableName: "Enumerable",
+    configurableName: "Configurable",
+    yes: "yes",
+    no: "no"
+  },
+  "ru": {
+    header: `Атрибуты свойства <code>${env.title}</code>`,
+    writableName: "Записываемое",
+    enumerableName: "Перечисляемое",
+    configurableName: "Настраиваемое",
+    yes: "да",
+    no: "нет"
+  },
+  "zh-CN": {
+    header: `<code>${env.title}</code> 的属性特性`,
+    writableName: "可写",
+    enumerableName: "可枚举",
+    configurableName: "可配置",
+    yes: "是",
+    no: "否"
+  },
+  "zh-TW": {
+    header: `<code>${env.title}</code> 的屬性特性`,
+    writableName: "可寫",
+    enumerableName: "可列舉",
+    configurableName: "可配置",
+    yes: "是",
+    no: "否"
+  }
+})
 
-switch(env.locale) {
-    case "fr":
-        header = 'Attributs de <code>' + env.title + '</code>';
-        writable_name = 'Écrivable';
-        enumerable_name = 'Énumérable';
-        configurable_name = 'Configurable';
-        yes = 'Oui';
-        no = 'Non';
-        break;
-    case "es":
-        header = 'Atributos de la propiedad <code>' + env.title + '</code>';
-        writable_name = 'Sobrescribir';
-        enumerable_name = 'Numerable';
-        configurable_name = 'Configurable';
-        yes = 'Sí';
-        no = 'No';
-        break;
-    case "ru":
-        header = 'Атрибуты свойства' + '<code>' + env.title + '</code>';
-        writable_name = 'Записываемое';
-        enumerable_name = 'Перечисляемое';
-        configurable_name = 'Настраиваемое';
-        yes = 'да';
-        no = 'нет';
-        break;
-    case "zh-CN":
-        header = '<code>' + env.title + '</code> 属性的属性特性：';
-        writable_name = 'writable';
-        enumerable_name = 'enumerable';
-        configurable_name = 'configurable';
-        yes = 'true';
-        no = 'false';
-        break;
-    case "ja":
-        header = '<code>' + env.title + '</code> のプロパティ属性';
-        writable_name = '書込可能';
-        enumerable_name = '列挙可能';
-        configurable_name = '設定可能';
-        yes = '可';
-        no = '不可';
-        break;
-    case "uk":
-        header = 'Атрибути поля <code>' + env.title + '</code>';
-        writable_name = 'Доступний для запису';
-        enumerable_name = 'Доступний для переліку';
-        configurable_name = 'Доступний для налаштування';
-        yes = 'так';
-        no = 'ні';
-        break;
-    default:
-        header = 'Property attributes of <code>' + env.title + '</code>';
-        writable_name = 'Writable';
-        enumerable_name = 'Enumerable';
-        configurable_name = 'Configurable';
-        yes = 'yes';
-        no = 'no';
-        break;
-}
-
-var writable = no;
-var enumerable = no;
-var configurable = no;
-
-if ($0 == 1) {
-    writable = yes;
-}
-if ($1 == 1) {
-    enumerable = yes;
-}
-if ($2 == 1) {
-    configurable = yes;
-}
+const writable = $0 == 1 ? text.yes : text.no;
+const enumerable = $1 == 1 ? text.yes : text.no;
+const configurable = $2 == 1 ? text.yes : text.no;
```

Comment:
I'm a bit not sure should we use **strict equality** here. Maybe `"1"` is used somewhere?

Reply:
Yes, let's be defensive and keep it as is.

However, could we refactor this so that `writable/enumerable/configurable` are just booleans, and the conversion to text happens below, i.e. separating logic from presentation?

```suggestion
const isWritable = $0 == 1;
const isEnumerable = $1 == 1;
const isConfigurable = $2 == 1;
```